merlin_models/tf/core.py

```diff
@@ -2217,7 +2226,7 @@ def train_step(self, inputs):
                 targets = None
 
             predictions = self(inputs, training=True)
-            loss = self.compute_loss(predictions, targets, training=True)
+            loss = self.compute_loss(predictions, targets, training=True, compute_metrics=True)
```

Comment:
Is it always going to compute metrics during training?

Reply:
Or if we have a separate list of `train_metrics` and `eval_metrics`, then if `train_metrics = [] / None` that would mean that no eval metrics should be processed during training (including the `pre_metrics` call)